### ISSUE TITLE
When will fileRegion be supported? 

### ISSUE THREAD

**Issue by weitingyuk** [netty/netty-incubator-transport-io_uring#210]
The io_uring transport does not currently support fileRegion, when will fileRegion be supported?  We are eager to try iouringFileRegion.

**Comment by weitingyuk** [comment]
@1Jo1 Cloud you mind help to see this issue?

**Comment by normanmaurer** [comment]
Seems like while sendfile is not directly supported by io_uring itself it is possible to kind of create the same functionality by using splice: https://lwn.net/Articles/810482/

**Comment by weitingyuk** [comment]
@normanmaurer I don't know much about the Linux kernel. Will netty io_uring consider using splice to implement fileRegion?

**Comment by weitingyuk** [comment]
I'm not sure there is some gap for netty io_uring  to implement fileRegion by using splice.

**Comment by normanmaurer** [comment]
Yes .... I just left the comment for the future so I dont forget. I am currently busy but might implement it once I have some cycles.

**Comment by weitingyuk** [comment]
We really want to try the file region in netty io_uring. If there is no plan for this currently, we can only abandon this plan, thanks very much.

**Comment by normanmaurer** [comment]
If someone implements it I am happy to review the code... I dont have the cycles atm to do it myself

**Comment by dreamlike-ocean** [comment]
Hello，Can we have an IoUringChunkedNioFile? I tried it and it doesn't seem difficult.

**Comment by franz1981** [comment]
hi @dreamlike-ocean I think this one to be a better fit outside Netty, which is just a "network" framework.
FileRegion is not indeed a good term because in Netty it implies zero-copy semantic, which is not happening here; I believe that having such async file abstraction in vertx, for example, to be super cool @vietj wdyt?

**Comment by normanmaurer** [comment]
@franz1981 actually FileRegion should do a zero copy (aka sendfile). We just never did it so far... It should be doable as io_uring supports splice.

**Comment by dreamlike-ocean** [comment]
> @franz1981 actually FileRegion should do a zero copy (aka sendfile). We just never did it so far... It should be doable as io_uring supports splice.

yes ，https://github.com/dreamlike-ocean/PanamaUring/blob/master/panama-uring/src/main/java/top/dreamlike/panama/uring/async/fd/AsyncSplicer.java

**Comment by normanmaurer** [comment]
Sure we love PRs

**Comment by dreamlike-ocean** [comment]
> Sure we love PRs

Hi! I have submitted a pull request for this requirement. Please  review it.Check if I have correctly understood this requirement.

**Comment by normanmaurer** [comment]
It' supported in the io_uring that we will be part of 4.2.

**Comment by vietj** [comment]
great work!
